inst/templates/get-started.Rmd

```diff
@@ -383,6 +383,49 @@ vroom::vroom_write(to_edit, "to_edit.csv", delim = ",")
 writexl::write_xlsx(to_edit, "to_edit.xlsx")
 ```
 
+### Calculate coverage statistics
+
+```{r}
+# Dataframe with TRUE suggested matches
+suggested <- to_edit %>%
+  filter(suggest_match)
+
+# Matched companies based on suggested matches
+matched_rows <- suggested %>% nrow()
+
+# Share of matched companies from total loanbook companies
+matched_share <- matched_rows / loanbook %>%
```

Comment:
I'm puzzled. I'm running this in my head and it doesn't work. Even if it works it's surprising so maybe best to rewrite it?

``` r
library(dplyr, warn.conflicts = FALSE)

suggested <- tribble(
  ~company_name, ~suggest_match,
  "abc",                   TRUE,
  "def",                   TRUE,
)

loanbook <- tribble(
  ~company_name, ~postcode,
  "abc",            "1234",
  "def",            "5678",
)
loanbook
#> # A tibble: 2 × 2
#>   company_name postcode
#>   <chr>        <chr>   
#> 1 abc          1234    
#> 2 def          5678

nrow(suggested) / loanbook
#> Error in FUN(left, right): non-numeric argument to binary operator
```

<sup>Created on 2023-03-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

No need for action just now. Let me run it and see what's going on.

Reply:
@kalashsinghal, 
I run the code and confirmed there is a bug here. 

See the [rendered version of this file](https://github.com/2DegreesInvesting/tilt.company.match/blob/100_coverage_statistics/inst/templates/get-started.md#calculate-coverage-statistics) (which we'll need to remove once we solve this issue). 

You'll see this:

```r
suggested <- to_edit %>%
  filter(suggest_match)

#### Number of matched companies based on suggested matches
matched_rows <- suggested %>% nrow()
matched_rows
#> [1] 3

#### Share of matched companies from total loanbook companies
matched_share <- matched_rows / loanbook %>%
  distinct(company_name) %>%
  nrow()
matched_share
#> [1] 0.375
```

The bug is now clear: `nrow()` -- the number of rows -- should be an integer so 0.375 is wrong.

Try to rewrite this code without the pipe:

```r
# Fails
step1 <- matched_rows / loanbook
step2 <- distinct(step1, company_name)
step3 <- nrow(step2)
```

The problem is in `step1`. Can you say in words what are you trying to do?

Reply:
@maurolepore My purpose is to calculate the share of matched companies from loanbook companies which have unique company names. Maybe this code snippet is more clear: 
```
total_loan <- loanbook %>%
  distinct(company_name) %>%
  nrow()
matched_share <- matched_rows / total_loan
matched_share
```
I will make this change after you will approve of it

Reply:
This is super clear and running it in my head works like charm :-) thanks!!!

Reply:
Hi @maurolepore Should I also render this rmd file?

Reply:
Clarity is crucial as it reduces the review-time and chances of introducing bugs. It's your call how you achieve clarity. In this particular case breaking up the pipe into its pieces exposed a bug. Did you notice? Your last code is not the same as before but clearer -- it's different code. 

```r
# Bad: Number / dataframe
matched_rows / loanbook %>%  # Bug here
  distinct(company_name) %>%
  nrow()

# Good: Number / number
matched_rows / nrow(distinct(loanbook, company_name))
```

Reply:
@maurolepore Thanks for this insight! :)